keanu-project/src/main/java/io/improbable/keanu/vertices/LogProbGraph.java

```diff
@@ -0,0 +1,53 @@
+package io.improbable.keanu.vertices;
+
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.vertices.dbl.Differentiable;
+import io.improbable.keanu.vertices.dbl.DoubleVertex;
+import io.improbable.keanu.vertices.dbl.KeanuRandom;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.Singular;
+
+import java.util.Map;
+
+/**
+ * A graph of vertices representing the computation of a log probability
+ */
+@Builder
+public class LogProbGraph {
+
+    /**
+     * A mapping from vertices to placeholders, which would have its values fed during execution.
```

Comment:
I think this comment goes outside the scope of this class. This is good information but it belongs with the code that uses this class in that way.